kof-operator/internal/controller/clusterdeployment_controller_test.go

```diff
@@ -302,17 +366,32 @@ var _ = Describe("ClusterDeployment Controller", func() {
 		It("should successfully reconcile the resource", func() {
 			By("Reconciling the created resource")
 			clusterDeployment := &kcmv1alpha1.ClusterDeployment{}
-			err := k8sClient.Get(ctx, clusterDeploymentNamespacedName, clusterDeployment)
+			err := k8sClient.Get(ctx, childClusterDeploymentNamespacedName, clusterDeployment)
 			Expect(err).NotTo(HaveOccurred())
 
 			_, err = controllerReconciler.Reconcile(ctx, reconcile.Request{
-				NamespacedName: clusterDeploymentNamespacedName,
+				NamespacedName: childClusterDeploymentNamespacedName,
 			})
 			Expect(err).NotTo(HaveOccurred())
 
-			remoteSecret := &coreV1.Secret{}
+			remoteSecret := &corev1.Secret{}
 			err = k8sClient.Get(ctx, remoteSecretNamespacedName, remoteSecret)
 			Expect(err).NotTo(HaveOccurred())
 		})
+
+		It("should create ConfigMap for child cluster", func() {
```

Comment:
Please check that update logic based on ClusterDeployment generation is working.